server/src/routes/connection.ts

```diff
@@ -0,0 +1,92 @@
+import { NextFunction, Request, Response, Router } from 'express';
+import createError from 'http-errors';
+import { ConnectionCreateInput, connectionService, mindmapService } from '../services';
+
+const router = Router();
+
+router.get('/:id', async (req: Request, res: Response, next: NextFunction) => {
+  const id = parseInt(req.params.id);
+
+  try {
+    const result = await connectionService.findOrThrow(id);
+    res.json(result);
+  } catch (err) {
+    next(createError(500, err));
+    return;
+  }
+});
+
+router.post('/', async (req: Request, res: Response, next: NextFunction) => {
+  const connection = req.body as ConnectionCreateInput;
+
+  const mindmapId = connection.MindmapId;
+  const connections = await mindmapService.getConnections(mindmapId);
+
+  for (let i = 0; i < connections.length; i++) {
+    if (connections[i].label === connection.label) {
```

Comment:
I would imagine that if the source and target id matches the connection would be the same? We can have the same label names in the mindmap